src/nl/hannahsten/texifyidea/refactoring/inlinefile/LatexInlineFileHandler.kt

```diff
@@ -0,0 +1,117 @@
+
+    override fun isEnabledForLanguage(language: Language?): Boolean {
+        return language == LatexLanguage
+    }
+
+    override fun canInlineElement(element: PsiElement?): Boolean {
+        return canInlineLatexElement(element)
+    }
+
+    override fun inlineElement(project: Project, editor: Editor?, element: PsiElement) {
+        // Resolve the file to be inlined
+        val inlineFile: LatexFile = resolveInlineFile(element) ?: return
+
+        val dialog = LatexInlineFileDialog(
+            project,
+            inlineFile,
+            getReference(element, editor),
+            editor != null
+        )
+
+        if (dialog.myOccurrencesNumber > 0) {
+            if (ApplicationManager.getApplication().isUnitTestMode) {
+                try {
+                    dialog.doAction()
+                }
+                finally {
+                    dialog.close(DialogWrapper.OK_EXIT_CODE, true)
+                }
+            }
+            else {
+                dialog.show()
+            }
+        }
+        else {
+            // TODO No ocurrences, what to do?
+        }
+    }
+}
+
+/**
+ * This is static only so that the unit tests can use it, and also since it can be static
+ */
+fun canInlineLatexElement(element: PsiElement?): Boolean {
+    val out = when (element) {
+        is LatexFile -> {
+            element.containingFile.isLatexFile()
+        }
+
+        is LatexCommandsImpl -> {
+            ((element as LatexCommandsImpl).getName() == "\\input" && (element as PsiElementBase).canNavigateToSource())
+        }
+
+        else -> true
+    }
+
+    return out
+}
+
+fun resolveInlineFile(element: PsiElement) = when (element) {
+    // If we tried to inline the file in file view, or the argument in `input`
+    is LatexFile -> element
+    // If we tried to inline the `input` command
+    is LatexCommandsImpl -> element.containingFile.parent?.findFile(
+        if ((element as LatexCommandsImpl).requiredParameters[0].matches(
+                ".*\\.\\w{3}$".toRegex()
+            )
+        ) (element as LatexCommandsImpl).requiredParameters[0]
+        else (element as LatexCommandsImpl).requiredParameters[0] + ".tex"
+    ) as? LatexFile
+
+    else -> null
+}
+
+/**
+ * Resolves the command that the inlining was called on. If called from file view then expected return is null
+ */
+fun getReference(
+    element: PsiElement,
+    editor: Editor?
+): PsiElement? {
+    return if (element is LatexCommandsImpl)
+        element
+    else if (editor != null) {
+        if (getElementAtCaret(editor)?.parent?.parent?.parent is LatexCommandsImpl)
+            getElementAtCaret(editor)?.parent?.parent?.parent
+        else if (getElementAtCaret(editor)?.parent?.parent?.parent?.parent?.parent is LatexCommandsImpl)
+            getElementAtCaret(editor)?.parent?.parent?.parent?.parent?.parent
+        else {
+            Log.warn("Could not find the command element")
```

Comment:
If you decide to put a warning, then at least include the full element text and what the warning is about to make debugging easier